modules/ui/fields/input.js

```diff
@@ -213,17 +219,20 @@ export function uiFieldText(field, context) {
             // don't override multiple values with blank string
             if (!val && Array.isArray(_tags[field.key])) return;
 
-            if (!onInput) {
-                if (field.type === 'number' && val) {
-                    var vals = val.split(';');
-                    vals = vals.map(function(v) {
-                        var num = parseFloat(v.trim(), 10);
-                        return isFinite(num) ? clamped(num) : v.trim();
-                    });
-                    val = vals.join(';');
-                }
-                utilGetSetValue(input, val);
+            var displayVal = val;
+            if (field.type === 'number' && val) {
+                var vals = val.split(';');
+                vals = vals.map(function(v) {
+                    v = v.trim();
+                    var num = parseLocaleFloat(v);
+                    if (!isFinite(num)) return v;
+                    num = parseFloat(num, 10);
+                    if (!isFinite(num)) return v;
+                    return clamped(num);
```

Comment:
see above. AFAICS this could be reduced to

```suggestion
                    var num = parseLocaleFloat(v);
                    return isFinite(num)) ? clamped(num) : v;
```